state/unit.go

```diff
@@ -272,17 +272,17 @@ func (u *Unit) SetAgentVersion(v version.Binary) (err error) {
 	if err = checkVersionValidity(v); err != nil {
 		return err
 	}
-	tools := &tools.Tools{Version: v}
+	result := &tools.Tools{Version: v}
```

Comment:
Same here - how about `newTools` instead?

Reply:
changing to 'versionedTool' :D